### PR TITLE
Abstract over dependency kind in Compile

### PR DESCRIPTION
**Pull request by Duhemm**
This follows up on my work to make it easier to add new dependency kind in sbt.

This is the last part that needs to be adapted to take advantage of the new infrastructure that has been introduced by :
- Abstract over dependency kind in Analysis (#1340)
- Don't hardcode relations in TextAnalysisFormat (#1572)

/cc @xeno-by

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

File: compile/inc/src/main/scala/sbt/inc/Analysis.scala [hunk 1]
```diff
@@ -5,6 +5,7 @@ package sbt
 package inc
 
 import xsbti.api.Source
+import xsbti.DependencyContext._
```
Comment (gkossakowski) [inline compile/inc/src/main/scala/sbt/inc/Analysis.scala:8]: Why is this needed? If nothing else is modified in the file I'm guessing this import is a dead code.

Comment (Duhemm) [inline compile/inc/src/main/scala/sbt/inc/Analysis.scala:8]: This file uses `DependencyByMemberRef` and `DependencyByInheritance` that were previously defined in the same package as `Analysis`. `DependencyByMemberRef` and `DependencyByInheritance` have been moved to an `enum` in the package `xsbti`, because the interface `AnalysisCallback` uses them.

**Comment by Duhemm** [comment]
Hey,
Any update about the status of this pull request ? Thanks !

**Comment by jsuereth** [comment]
@gkossakowski is this one good to go?

**Comment by gkossakowski** [comment]
LGTM

Yes, merging.
